core/src/test/scala/integration/kafka/api/BaseConsumerTest.scala

```diff
@@ -130,6 +137,12 @@ abstract class BaseConsumerTest extends IntegrationTestHarness {
     }
   }
 
+
+  protected def createConsumerWithGroupId(groupId: String): KafkaConsumer[Array[Byte], Array[Byte]] = {
+    consumerConfig.setProperty(ConsumerConfig.GROUP_ID_CONFIG, groupId)
```

Comment:
It's a little surprising to find that we mutate `consumerConfig` here. So if you called `createConsumerWithGroupId` first and then `createConsumer`, the first call would affect the second call. Maybe instead we could duplicate `consumerConfig` and override group id?

Reply:
Good call. I'll make use of the `configOverrides` var in `createConsumer()`